### PR TITLE
[Distributed] add distributed evaluation.

### PR DESCRIPTION
**Pull request by zheng-da**
## Description
This PR adds distributed evaluation in the distributed GraphSage example.

It extends DistTensor to create node/edge data in KVStore directly.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: examples/pytorch/graphsage/experimental/train_dist.py [hunk 1]
```diff
@@ -43,6 +43,63 @@ def sample_blocks(self, seeds):
...
+        layers.
+        """
+        # During inference with sampling, multi-layer blocks are very inefficient because
+        # lots of computations in the first few layers are repeated.
+        # Therefore, we compute the representation of all nodes layer by layer.  The nodes
+        # on each layer are of course splitted in batches.
+        # TODO: can we standardize this?
+        nodes = dgl.distributed.node_split(np.arange(g.number_of_nodes()),
+                                           g.get_partition_book(), force_even=True)
+        y = dgl.distributed.DistTensor(g, (g.number_of_nodes(), self.n_hidden), th.float32, 'h',
```
Comment (classicsong) [inline examples/pytorch/graphsage/experimental/train_dist.py:68]: When will we free this tensor?

File: python/dgl/distributed/graph_partition_book.py [hunk 2]
```diff
@@ -623,6 +623,9 @@ def partid(self):
         """
         return self._partid
 
+NODE_PART_POLICY = 'node'
```
Comment (classicsong) [inline python/dgl/distributed/graph_partition_book.py:626]: Why we need this?

Comment (zheng-da) [reply python/dgl/distributed/graph_partition_book.py:626]: we need different partition policies for nodes and edges. i just create "constants" for the two policies.

File: examples/pytorch/graphsage/experimental/train_dist.py (hunk 1 above)
Comment (zheng-da) [reply examples/pytorch/graphsage/experimental/train_dist.py:68]: right now, it doesn't. i'll have another PR to free distributed tensors after DistTensor is destroyed.

File: python/dgl/distributed/dist_tensor.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            self.kvstore.delete_data(self._name)
+
+    def __getitem__(self, idx):
+        idx = utils.toindex(idx)
+        idx = idx.tousertensor()
+        return self.kvstore.pull(name=self._name, id_tensor=idx)
+
+    def __setitem__(self, idx, val):
+        idx = utils.toindex(idx)
+        idx = idx.tousertensor()
```
Comment (Lyken17) [inline python/dgl/distributed/dist_tensor.py:102]: Why there are many 

```
idx = utils.toindex(idx)
idx = idx.tousertensor()
```

first converts idx to `Index` then back to `Tensor`?

**Pull request merged**